app/assets/javascripts/models/beta/projectBoard.js

```diff
@@ -68,44 +68,72 @@ export const toggleColumn = (projectBoard, column, callback) => {
 }
 
 // Drag And drop utils
+const calculateNewPosition = (aboveStory, bellowStory, storyState) => {
+  const aboveStoryState = aboveStory?.state;
+  const bellowStoryState = bellowStory?.state;
+  const aboveStoryPosition = aboveStory?.newPosition;
+  const bellowStoryPosition = bellowStory?.newPosition;
+
+  if (!aboveStory) return 1;
+  if (!bellowStory) return Number(aboveStoryPosition) + 1;
+  if (aboveStoryState === storyState && bellowStoryState !== storyState) return Number(aboveStoryPosition) + 1;
+  if (bellowStoryState === storyState && aboveStoryState !== storyState) return Number(bellowStoryPosition) - 1;
```

Comment:
My mistake, the `calculateNewPosition` function is a copy of the existing `calculatePosition`, I ended up not noticing the typo, but I have already corrected both functions.

About the cast for Number, it follows the same idea of the typo, for some reason the property `position` is received in String format and needs to go through the cast, but I will change it, since `newPosition` does not need this.